src/components/publishing/display/canvas/slideshow.tsx

```diff
@@ -0,0 +1,217 @@
+import Colors from "../../../../assets/colors"
+import { crop } from "../../../../utils/resizer"
+import { pMedia } from "../../../helpers"
+import Icon from "../../../icon"
+import Fonts from "../../fonts"
+
+interface SlideshowProps {
+  unit: any
+  disclaimer: any
+  containerWidth: number
+}
+
+class DisplayCanvasSlideshow extends React.Component<SlideshowProps, any> {
+  private slider: any
+
+  constructor(props) {
+    super(props)
+    this.state = { isOnTitle: true }
+  }
+
+  renderTitleCard() {
+    const { children, disclaimer, containerWidth, unit } = this.props
+    const hasCaptions = compact(map(unit.assets, "caption")).length > 0
+    return (
+      <Title containerWidth={containerWidth}>
+        {children}
+        {hasCaptions && <Disclaimer>{disclaimer}</Disclaimer>}
+      </Title>
+    )
+  }
+
+  renderSlides = () => {
+    const { unit, containerWidth } = this.props
+    return unit.assets.map((image, i) => {
+      return (
+        <Slide key={i} className={i === 0 && "title-card"} containerWidth={containerWidth}>
+          {i === 0 && containerWidth >= 900 && this.renderTitleCard()}
+          <div>
+            <Image
+              src={crop(image.url, { width: 780, height: 460 })}
+              alt={image.caption || ""}
+              containerWidth={containerWidth}
+            />
+            {image.caption && <Caption>{image.caption}</Caption>}
+          </div>
+        </Slide>
+      )
+    })
+  }
+
+  onSlideChange = currentSlide => {
+    this.setState({ isOnTitle: currentSlide === 0 })
+  }
+
+  render() {
+    const { containerWidth } = this.props
+    const sliderSettings = {
+      dots: false,
+      lazyLoad: false,
+      infinite: false,
+      variableWidth: true,
+      centerMode: true,
+      nextArrow: <RightArrow containerWidth={containerWidth} onRightArrow={() => this.slider.slickGoTo(0)} />,
+      prevArrow: <LeftArrow containerWidth={containerWidth} isOnTitle={this.state.isOnTitle} />,
+      initialSlide: 0,
+      afterChange: currentSlide => {
+        this.onSlideChange(currentSlide)
+      },
+      responsive: [
+        {
+          breakpoint: 900,
+          settings: {
+            infinite: true,
+          },
+        },
+      ],
+    }
+
+    return (
+      <SliderContainer containerWidth={containerWidth}>
+        <Slider {...sliderSettings} ref={slider => (this.slider = slider)}>
+          {this.renderSlides()}
+        </Slider>
+        {containerWidth < 900 && this.renderTitleCard()}
+      </SliderContainer>
+    )
+  }
+}
+
+const LeftArrow = props => {
+  return (
+    <NavArrow
+      containerWidth={props.containerWidth}
+      direction="left"
+      onClick={props.onClick}
+      isVisible={!props.isOnTitle}
+    >
+      <Icon name="chevron-left" color="black" fontSize="24px" />
+    </NavArrow>
+  )
+}
+
+const RightArrow = props => {
+  const onClick = props.currentSlide + 1 === props.slideCount ? props.onRightArrow : props.onClick
```

Comment:
Can you explain the reasoning for rewriting the infinite scroll functionality in desktop view? I believe it has to do with the title card, but I'm not understanding why we can't use `infinite: true` anyway.

Reply:
VariableWidth (needed for the title card) creates some bugs when combined with infinite mode -- https://github.com/kenwheeler/slick/issues/1207

We don't need VariableWidth in mobile, so it is ok to use infinite there (also just realizing I forgot to remove variableWidth in the responsive section).